NGitLab.Mock/Repository.cs

```diff
@@ -636,6 +636,43 @@ public bool Rebase(UserRef user, string sourceBranch, string targetBranch)
             return true;
         }
 
+        internal bool HasConflicts(UserRef user, string sourceBranch, string targetBranch)
+        {
+            if (user is null)
+                throw new ArgumentNullException(nameof(user));
+            if (string.IsNullOrEmpty(sourceBranch))
+                throw new ArgumentException("Cannot be null or empty", nameof(sourceBranch));
+            if (string.IsNullOrEmpty(targetBranch))
+                throw new ArgumentException("Cannot be null or empty", nameof(targetBranch));
+
+            var repo = GetGitRepository();
+
+            var head = repo.Head;
+            var branch = repo.Branches[sourceBranch];
+            var upstream = repo.Branches[targetBranch];
+
+            _ = repo.Network.Remotes[targetBranch];
+
+            var signature = new Signature(user.UserName, user.Email, DateTimeOffset.UtcNow);
+            var options = new MergeOptions
+            {
+                CommitOnSuccess = false,
+                FastForwardStrategy = FastForwardStrategy.NoFastForward,
+            };
+
+            Commands.Checkout(repo, upstream);
+
+            var resetTip = upstream.Tip;
+
+            var mergeResult = repo.Merge(branch.Tip, signature, options);
+            var result = mergeResult.Status == MergeStatus.Conflicts;
+
+            repo.Reset(ResetMode.Hard, resetTip);
+            Commands.Checkout(repo, head);
```

Comment:
Do you think we can use `new CheckoutOptions { CheckoutModifiers = CheckoutModifiers.Force }` and remove the previous `Reset`? (@louis-z)

Reply:
If I don't use `Reset`, the merging operation is not aborted

![image](https://user-images.githubusercontent.com/23695924/218668001-ef6f8842-3ac0-4149-a4ec-eaf64ae4f82c.png)

Reply:
You are right